uui-core/src/types/props.ts

```diff
@@ -128,6 +127,136 @@ export interface IBasicPickerToggler extends IDropdownToggler {
     onClear?(e?: any): void;
 }
 
+/** Holds parent info for data rows */
+export interface DataRowPathItem<TId, TItem> {
```

Comment:
https://github.com/epam/UUI/blob/5c9956fff284dc91f0c2547991d139d378660cf2/uui-core/src/types/props.ts#L6C40-L6C40
https://github.com/epam/UUI/blob/5c9956fff284dc91f0c2547991d139d378660cf2/uui-core/src/types/dataSources.ts#L4C13-L4C13

Because, otherwise, a cyclic import is present in those files.

Reply:
@AlekseyManetov, unfortunately, it will be the dependency cycle again. 

https://github.com/epam/UUI/blob/ac52fddc4cebdbe3a9b6eacd5f9a4ca4847c8f03/uui-core/src/types/components/PickerBase.ts#L2C4-L2C4
`PickerBase` types require types from `props.ts`. `DataRowPathItem` is used in types, used not only in pickers, but in all dataSources consumers, such as `DataRowProps.path`. So, moving `DataRowPathItem` to `pickers.ts` and exporting them to `props.ts` or `dataSources.ts` is pretty wrong. 

So, I've done some refactoring. Please, take a look at it. Thanks.

cc @jakobz